activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -242,7 +242,16 @@ def references!(*table_names) # :nodoc:
     #   Model.select(:field).first.other_field
     #   # => ActiveModel::MissingAttributeError: missing attribute: other_field
     def select(*fields)
-      return super if block_given?
+      if block_given?
+        if fields.any?
+          ActiveSupport::Deprecation.warn(<<-WARNING.squish)
+            When select is called with a block, it ignores other arguments. This behavior is now deprecated and will result in an ArgumentError in Rails 5.1. You can safely remove the arguments to resolve the deprecation warning because they do not have any effect on the output of the call to select method with a block.
+          WARNING
```

Comment:
sorry one last very minor comment, can you update the end to say `of the call to the select method with a block`? other than that 👍